### PR TITLE
Custom underlying services

### PR DESCRIPTION
**Pull request by eguzki**
Related issue: https://github.com/Kuadrant/kuadrant-operator/issues/164

### CODE REVIEW

**Comment by eguzki** [comment]
There are still some `TODO`s and empty sections. Looking for some first comments, though.

File: rfcs/0000-custom-underlying-services.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+- It is reasonably clear how the feature would be implemented.
+- How error would be reported to the users.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+TODO
```
Comment (pehala) [inline rfcs/0000-custom-underlying-services.md:106]: One potential drawback is that we do not want users to change some settings. From the top of my head, I can think of Authorino `spec.clusterWide` and `authConfigLabelSelectors`, which could mean that the user could easily end up with AuthPolicy that will be reconciled by Kuadrant but won't be picked up in Authorino.

File: rfcs/0000-custom-underlying-services.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+One kuadrant control plane instance is represented by one Kuadrant custom resource. The main idea is to wire the underlaying services backing kuadrant with the kuadrant instance within the Kuadrant custom resource spec.
+
+```yaml
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: <kuadrant-name>
+  namespace: <kuadrant-namespace>
+spec:
+  authorinoTargetRef:
```
Comment (guicassolato) [inline rfcs/0000-custom-underlying-services.md:32]:
```suggestion
  authorinoRef:
```

File: rfcs/0000-custom-underlying-services.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      name: <authorino-name>
+      namespace: <authorino-namespace>
+  limitadorTargetRef:
```
Comment (guicassolato) [inline rfcs/0000-custom-underlying-services.md:35]:
```suggestion
  limitadorRef:
```

File: rfcs/0000-custom-underlying-services.md (hunk 1 above)
Comment (eguzki) [reply rfcs/0000-custom-underlying-services.md:106]: Kuadrant should reconcile those changes in the external services, right? More specifically, when the Authorino CR is (externally) updated and the label selectors are updated, the AuthConfig CR's managed by kuadrant should reconcile to match.

**Comment by eguzki** [comment]
Updating with my latest changes and handing over to anyone from the team

**Pull request closed without merging**